integration_tests/src/main/python/spark_init_internal.py

```diff
@@ -81,7 +82,23 @@ def pyspark_ready():
         return False
 
 
+def global_init():
+    logging.info("Executing global initialization tasks before test launches")
+    create_tmp_hive()
+
+def create_tmp_hive():
+    path = '/tmp/hive'
+    mode = stat.S_IRWXU | stat.S_IRWXG | stat.S_IRWXO
+    logging.info(f"Creating directory {path} with permissions {oct(mode)}")
+    os.makedirs(path, mode, exist_ok=True)
+    os.chmod(path, mode)
+
 def pytest_sessionstart(session):
+    # initializations that must happen globally once before tests start
+    # if xdist in the coordinator, if not xdist in the pytest process
```

Comment:
I don't understand the comment, with xdist it automatically calls global_init?

Reply:
when pytest is run with xdist global_init is called only in the coordinator process to ensure it's called only once. Without xdist global_init is called in the same process that is running the tests.